pkg/registry/core/service/strategy.go

```diff
@@ -121,6 +120,10 @@ func dropServiceDisabledFields(newSvc *api.Service, oldSvc *api.Service) {
 	if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && !serviceIPFamilyInUse(oldSvc) {
 		newSvc.Spec.IPFamily = nil
 	}
+	// Drop TopologyKeys if ServiceTopology is not enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ServiceTopology) && !topologyKeysInUse(oldSvc) {
+		newSvc.Spec.TopologyKeys = nil
```

Comment:
Drop `TopologyKeys` if `ServiceTopology` is not enabled  to prevent compatibility issues in some edge case. (e.g. create Service with TopologyKeys when `ServiceTopology` is not enabled, then upgrade master with `ServiceTopology` enabled, but Service with TopologyKeys have already stored in etcd and not been validated, if it's invalid, some weird thing might happen.